code/modules/research/xenobiology/crossbreeding/stabilized.dm

```diff
@@ -23,44 +22,37 @@ Stabilized extracts:
 	qdel(linked_effect)
 	return ..()
 
+/// Returns the mob that is currently holding us if we are either in their inventory or a backpack analogue
+/obj/item/slimecross/stabilized/proc/get_held_mob()
+	if(!loc)
+		return FALSE
+	if(isliving(loc))
+		return loc
+	// Snowflake check for modsuit backpacks, which should be valid but are 3 rather than 2 steps from the owner
+	if(istype(loc, /obj/item/mod/module/storage))
+		var/obj/item/mod/module/storage/mod_backpack = loc
+		var/mob/living/modsuit_wearer = mod_backpack.mod?.wearer
+		return modsuit_wearer ? modsuit_wearer : FALSE
+	var/nested_loc = loc.loc
+	if (isliving(nested_loc))
+		return nested_loc
+	return FALSE
+
 /obj/item/slimecross/stabilized/process()
-	var/humanfound = null
-	if(ishuman(loc))
-		humanfound = loc
-	if(ishuman(loc.loc)) //Check if in backpack.
-		humanfound = (loc.loc)
-	for(var/atom/storage_loc as anything in get_storage_locs(src))
-		if(ishuman(storage_loc))
-			humanfound = storage_loc
-			break
-		if(ishuman(storage_loc.loc))
-			humanfound = storage_loc.loc
-			break
-		for(var/atom/storage_loc_storage_loc as anything in get_storage_locs(storage_loc))
-			if(ishuman(storage_loc_storage_loc))
-				humanfound = storage_loc_storage_loc
-				break
-	for(var/atom/loc_storage_loc as anything in get_storage_locs(loc))
-		if(ishuman(loc_storage_loc))
-			humanfound = loc_storage_loc
-			break
-	if(!humanfound)
+	var/mob/living/holder = get_held_mob()
+	if(!holder)
```

Comment:
I'd rather `get_held_mob()` return either `null` or the valid `/mob/living` just so that we can certainly say "we are going to either get a valid mob reference or absolutely nothing" rather than deal with "we can either get a reference or the number zero". it's okay to explicitly write `return null` and then do the `isnull()` check.